### DISCUSSION TITLE
Can't trigger form submit on blur of modifying an input elelement

### DISCUSSION THREAD

**Discussion by keypuncherlabs** [mantinedev/mantine#2939] [Q&A] [answered]
### What package has an issue

@mantine/form

### Describe the bug

I'm currently at a loss trying to trigger a form submit event on the form created by use-form.

My use case is that I would simply like the form to submit when a user clicks anywhere outside the current form input element.

I tried attaching a ref on the Form itself and noticed that it does not support that property.

Is this something that would be possible for me to add in a PR or is there already a workaround for this?  I'm sure not everyone wants a 'Submit' button in every form to trigger the submit.

### What version of @mantine/hooks page do you have in package.json?

5.7.1

### Do you know how to fix the issue

Yes

### Are you willing to participate in fixing this issue and create a pull request with the fix

Yes

### Possible fix

I think it would be nice to expose a 'submit()' method on the instantiated 'form' instance.

Example:
```
import { useForm } from '@mantine/form';
const form = useForm({
    initialValues: {
      ...
    },
    validate: {
      ...
    },
  });
<TextInput onBlur={form.submit()} />
```

**Comment by rtivital** [comment, marked as answer]
You can save submit function in a variable and call it:

```
const submitForm = form.onSubmit(() => {});

<TextInput onBlur={() => submitForm()} />
```

**Reply by keypuncherlabs** [reply]
Thanks @rtivital, that worked for me as well!  Really great design and UX in this library, I'm really enjoying working with it. Nice work!

**Reply by rtivital** [reply]
Thanks for kind words!

**Comment by cwbuecheler** [comment]
@rtivital - We're running into a minor issue that makes me a touch nervous. We'd like to do a few things onBlur with a certain textfield. However, we're also using `useForm` to control the field, like this:

```javascript
<TextInput
  disabled={isDeviceHcidInputDisabled}
  label="Device HCID"
  placeholder="Device HCID"
  ref={ref}
  required
  {...form.getInputProps('deviceHCID')}
/>
```

if we put an `onBlur` above the `getInputProps` line, it won't trigger, which I assume means that the `getInputProps` spread contributes its own `onBlur` function. If we put our `onBlur` at the bottom, it works ... which is great, but I assume that means I'm overwriting `useForms`'s function. Is that an okay thing to do?

Thanks!

**Reply by rtivital** [reply]
getInputProps provides onBlur callback, if you want to override it set onBlur prop after spreading getInputProps. onBlur handler is used to validate fields on blur

**Reply by cwbuecheler** [reply]
I thought that might be the case. Thanks!

**Reply by pc-erin** [reply]
@rtivital Can we call onBlur in our own handler if we want to keep the default validation behavior?

```tsx
<TextInput
  disabled={isDeviceHcidInputDisabled}
  label="Device HCID"
  placeholder="Device HCID"
  ref={ref}
  required
  {...form.getInputProps('deviceHCID')}
  onBlur={(e) => {
    form.getInputProps('deviceHCID').onBlur(e)
    // do our own stuff
  }}
/>
```

EDIT: Thanks, was kinda on autopilot earlier

**Reply by rtivital** [reply]
```ts
onBlur={event => {
  form.getInputProps('a').onBlur(event);
}}
```
